pkg/plugin/settings.go

```diff
@@ -13,24 +13,29 @@ import (
 
 // Settings - data loaded from grafana settings database
 type Settings struct {
-	Server             string `json:"server,omitempty"`
-	Port               int64  `json:"port,omitempty"`
-	Path               string `json:"path,omitempty"`
-	Username           string `json:"username,omitempty"`
-	DefaultDatabase    string `json:"defaultDatabase,omitempty"`
-	InsecureSkipVerify bool   `json:"tlsSkipVerify,omitempty"`
-	TlsClientAuth      bool   `json:"tlsAuth,omitempty"`
-	TlsAuthWithCACert  bool   `json:"tlsAuthWithCACert,omitempty"`
-	Password           string `json:"-,omitempty"`
-	TlsCACert          string
+	Host     string `json:"host,omitempty"`
+	Port     int64  `json:"port,omitempty"`
+	Protocol string `json:"protocol"`
+	Secure   bool   `json:"secure,omitempty"`
+	Path     string `json:"path,omitempty"`
+
+	InsecureSkipVerify bool `json:"tlsSkipVerify,omitempty"`
+	TlsClientAuth      bool `json:"tlsAuth,omitempty"`
+	TlsAuthWithCACert  bool `json:"tlsAuthWithCACert,omitempty"`
 	TlsClientCert      string
+	TlsCACert          string
 	TlsClientKey       string
-	Secure             bool            `json:"secure,omitempty"`
-	Timeout            string          `json:"timeout,omitempty"`
-	QueryTimeout       string          `json:"queryTimeout,omitempty"`
-	Protocol           string          `json:"protocol"`
-	CustomSettings     []CustomSetting `json:"customSettings"`
-	ProxyOptions       *proxy.Options
+
+	Username string `json:"username,omitempty"`
+	Password string `json:"-,omitempty"`
+
+	DefaultDatabase string `json:"defaultDatabase,omitempty"`
+
+	DialTimeout  string `json:"dialTimeout,omitempty"`
```

Comment:
Do you think we can introduce a sort of backward compatibility for `timeout` field? We don't need it, but would be nice to make users keep their configuration unchanged.

Reply:
Added this + tests for `Host` (`Server`) and `DialTimeout` (`Timeout`)